src/openapi/streaming/serializer/serializer-protobuf.js

```diff
@@ -0,0 +1,153 @@
+wrappers.register(protobuf.wrappers);
+
+function createRootSchema() {
+    let schemas = protobuf.Root.fromJSON(protobuf.common['google/protobuf/wrappers.proto'], protobuf.root);
+    schemas = protobuf.Root.fromJSON(protobuf.common['google/protobuf/timestamp.proto'], schemas);
+    return schemas;
+}
+
+/**
+ * Protobuf Serialization.
+ * Constructor name parameter has no other purpose then identification.
+ * @constructor
+ */
+function SerializerProtobuf(name) {
+    this.name = name;
+
+    /**
+     * Url to schema name map.
+     */
+    this.schemasMap = {};
+
+    /**
+     * Processed all meta fields of decoded message type.
+     * @type {MetaProtobuf}
+     */
+    this.metaProcessor = new MetaProcessor();
+}
+
+SerializerProtobuf.prototype.getSchemaType = function(schemaName, typeName) {
+    const schemas = this.schemasMap[schemaName];
+    return schemas && schemas.root.lookup(typeName);
+};
+
+SerializerProtobuf.prototype.getSchema = function(name) {
+    return this.schemasMap[name];
+};
+
+SerializerProtobuf.prototype.getSchemaNames = function() {
+    return Object.keys(this.schemasMap);
+};
+
+/**
+ * Parses and adds schema to local schema map.
+ * @param {String} schemaData - The schema data, not parsed, in raw, string format.
+ * @param {String} name - The schema name, under witch it will be saved in schema map.
+ * @return {boolean} - Returns true if there were no issues, false otherwise.
+ */
+SerializerProtobuf.prototype.addSchema = function(schemaData, name) {
+    if (this.schemasMap[name]) {
+        // Schemas for this name already exist.
+        return true;
+    }
+
+    let schema = createRootSchema();
+    try {
+        schema = protobuf.parse(schemaData, schema.root, { keepCase: true });
+    } catch (e) {
+        log.error(LOG_AREA, 'Schema parsing failed', {
+            error: e.message,
+            name,
+        });
+
+        return false;
+    }
+
+    this.schemasMap[name] = schema;
+    return true;
+};
+
+/**
+ * Parse data using given schema. Data should be in base64 format.
+ * @param {String} data - The data to parse. Data should be in base64 format.
+ * @param {String} schemaName - The name of a schema to be used for parsing.
+ * @return {*} - Result of parsing, if successful. Returns null if parsing fails or there is no data.
+ */
+SerializerProtobuf.prototype.parse = function(data, schemaName) {
+    const schemas = this.getSchema(schemaName);
+
+    if (!schemas || !data) {
+        return null;
+    }
+
+    const rootTypeName = schemas.root.getOption('saxobank_root');
+
+    if (!rootTypeName) {
+        log.error('Parsing failed. Missing root message name', rootTypeName);
+        return null;
+    }
+
+    const rootType = this.getSchemaType(schemaName, rootTypeName);
+
+    if (!rootType) {
+        log.error('Parsing failed. Root type not found. Name: ', rootTypeName);
+        return null;
+    }
+
+    const byteArray = new Uint8Array(protobuf.util.base64.length(data));
+
+    try {

```

Comment:
Try and catch slows down things a little bit. 
What do you think, should we keep it, or allow protobufjs errors to go to the top? 

It's performance critical method, I think we should get rid of.

Reply:
I'll check using some unit tests. And derive some tangible cost of try/catch.